src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -144,27 +151,89 @@ private void initialize() {
       public void actionPerformed(ActionEvent e) {
         UserPanel.this.getAllUsers(listModel);
       }
+    });   
+
+     userSignInButton.addActionListener(new ActionListener() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+       /* final String s = (String) JOptionPane.showInputDialog(
+            UserPanel.this, "Enter user name:", "Continue", JOptionPane.PLAIN_MESSAGE,
+            null, null, "");
+
+	final String p = (String) JOptionPane.showInputDialog(
+            UserPanel.this, "Enter Password:", "Sign-In", JOptionPane.PLAIN_MESSAGE,
+            null, null, "");
+*/
+	Login login = new Login(clientUser);
+	loginDialog.setVisible(true);
```

Comment:
Does this call block until the dialog is no longer visible? Otherwise it seems like the get* calls on the next line wouldn't necessarily return anything useful.